### ISSUE TITLE
Odd VM bug concerning nnkExprColonExpr

### ISSUE THREAD

**Issue by dom96** [nim-lang/Nim#1162]
```nimrod
import macros

proc processBody(node, retFutureSym: PNimrodNode,
                 subtypeName: string,
                 exceptBranch: PNimrodNode): PNimrodNode {.compileTime.} =
  echo("Processing ", node.kind)

  result = node
  case node.kind
  of nnkTryStmt:
    result = newNimNode(nnkStmtList)
    proc processForTry(i: var int, res: PNimrodNode, level: int) {.compileTime.} =
      var processed = processBody(node[0][i], retFutureSym, subtypeName, node[1])

    var i = 0
    processForTry(i, result, 0)
    return
  of nnkTableConstr:
    for i in 0 .. <result.len:
      echo(result[i].kind)
      result[i] = processBody(result[i], retFutureSym, subtypeName, exceptBranch)
    echo("Done")
    return
  of nnkExprColonExpr:
    echo(result.len)
    for i in 0 .. <result.len:
      echo(result[i].kind)
      result[i] = processBody(result[i], retFutureSym, subtypeName, exceptBranch)

  else: discard

  for i in 0 .. <result.len:

    result[i] = processBody(result[i], retFutureSym, subtypeName, exceptBranch)

when isMainModule:
  macro foo: stmt =
    var node = newNimNode(nnkTableConstr)
    var exp = newNimNode(nnkExprColonExpr)
    exp.add newStrLitNode("Content-type")
    exp.add newStrLitNode("text/html")
    node.add exp
    discard processBody(node, nil, "", nil)

  foo()
```
I would expect all of the children of the nnkExprColonExpr to be iterated over. It seems that the first string literal node is skipped.

```
Processing nnkTableConstr
nnkExprColonExpr
Processing nnkStrLit
Done
```

Commenting out the line in ``processForTry`` results in the expected output of:

```
Processing nnkTableConstr
nnkExprColonExpr
Processing nnkExprColonExpr
2
nnkStrLit
Processing nnkStrLit
nnkStrLit
Processing nnkStrLit
Processing nnkStrLit
Processing nnkStrLit
Done
```

**Comment by jangko** [comment]
this one already fixed, no longer valid issue for nim 0.12.0

**Comment by reactormonk** [comment]
Can confirm.
